extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java

```diff
@@ -156,20 +152,46 @@ private String parseFullFile(FileItem file, String baseuri) {
 
     }
 
-    private String getFormatFromUrl(String url)
-            throws IOException, MalformedURLException {
+    private String getFormatFromUrl(String url) throws IOException, MalformedURLException {
         StringBuffer response = new StringBuffer();
         URL obj = new URL(url);
-        HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+        HttpURLConnection con;
+
+        if (url.substring(0, 5).toLowerCase().equals("https")) {
+            con = (HttpsURLConnection) obj.openConnection();
+        } else {
+            con = (HttpURLConnection) obj.openConnection();
+        }
```

Comment:
This can be improved. Either `obj.getProtocol().equals("https")` (or `equalsIgnoreCase`), or `url.charAt(4) == 's'`. But then again, why cast it to `HttpsURLConnection` in the first place? [`HttpsURLConnection`](https://docs.oracle.com/javase/8/docs/api/index.html?javax/net/ssl/HttpsURLConnection.html) is a subclass of `HttpURLConnection`. You initialize a `HttpURLConnection` declared object, so the specifics of the `Https` class is not used anyway.